src/backend/distributed/utils/resource_lock.c

```diff
@@ -1196,3 +1199,147 @@ AcquireDistributedLockOnRelations(List *relationIdList, LOCKMODE lockMode)
 		SendCommandToWorker(nodeName, nodePort, lockCommand);
 	}
 }
+
+
+/*
+ * AcquireDistributedLockOnRelations filters relations before passing them to
+ * AcquireDistributedLockOnRelations_Internal to acquire the locks.
+ *
+ * Only tables, views, and foreign tables can be locked with this function. Other relations
+ * will cause an error.
+ *
+ * Skips non distributed relations.
+ * configs parameter is used to configure what relation will be locked and if the lock
+ * should throw an error if it cannot be acquired immediately
+ */
+void
+AcquireDistributedLockOnRelations(List *relationList, LOCKMODE lockMode, uint32 configs)
+{
+	if (!ClusterHasKnownMetadataWorkers() || !EnableMetadataSync || !EnableDDLPropagation)
+	{
+		return;
+	}
+
+	List *distributedRelationList = NIL;
+
+	/*
+	 * this is used to enforce the lock order:
+	 * [...A], [...B]
+	 * Where:
+	 * A: Excplicitly locked relations
```

Comment:
minor: typo Excplicitly


Also, `Relations which depend on` -> maybe `Relations that has referencing foreign key relationship` or such